kmir/src/tests/unit/test_linker.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, NamedTuple
+
+import pytest
+
+if TYPE_CHECKING:
+    from typing import Final
+
+
+class _TestData(NamedTuple):
+    symbol: str
+    name: str
+    expected_demangled: str
+    expected_qualified: str
+
+
+TEST_DATA: Final[tuple[_TestData, ...]] = (
+    _TestData(
+        symbol='_ZN14small_test_lib3add17h67a713ff87afe55cE',
+        name='add',
+        expected_demangled='small_test_lib::add::h67a713ff87afe55c',
+        expected_qualified='small_test_lib::add',
+    ),
```

Comment:
We could add a (synthetic) test using the names from the comment in `mono_item_fn_name` to document behaviour.